__tests__/app.test.js

```diff
@@ -79,6 +79,49 @@ describe("far-han-news tests", () => {
           });
         });
     });
+    test("GET 200: should accept topic as a filter", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(1);
+          expect(articles[0]).toHaveProperty("topic", "cats");
+        });
+    });
+    test("GET 200: should sort by specified column and order", () => {
+      return request(app)
+        .get("/api/articles?sort_by=comment_count&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeSortedBy("comment_count", { coerce: true });
+        });
+    });
+    test("GET 404: should return an error if passed passed a topic with no results", () => {
+      return request(app)
+        .get("/api/articles?topic=bananas&order=asc")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not found");
+        });
+    });
+    test('GET 400:should return an error if a forbidden sort_by is used', () => {
+      return request(app)
+      .get("/api/articles?sort_by=bananas")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+    });
+    test('GET 400: should return an error if a forbidden order is used', () => {
+      return request(app)
+      .get("/api/articles?order=bananas")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+    });
```

Comment:
What happens if there is a valid sort/existing topic with no articles? It should be a 200, like when you get an article with no comments. Not an error, but should be tested that it's happy on the 200.

Reply:
I see, just noticed the topics table